src/Util/ClassSourceManipulator.php

```diff
@@ -71,6 +72,7 @@ public function __construct(string $sourceCode, bool $overwrite = false, bool $u
         $this->printer = new PrettyPrinter();
 
         $this->setSourceCode($sourceCode);
+        $this->useAttributes = $useAttributes;
```

Comment:
This is getting complex! A few things:

1) Rename this to `bool $preferPhp8Attributes` - it seems more clear to me and differentiates it from `$useAnnotations`. Really, `$useAnnotations` could be renamed to `$shouldAnnotationProperties`, but let's not do that now.

2) This class is instantiated in multiple places. Unfortunately, we're going to need to make sure we pass in this new flag in all of those places (try running `git grep 'new ClassSourceMan'`).

Reply:
After more thought, I think we should NOT pass this argument to this class at all. We should not try to pass a flag to the constructor that says that we "prefer attributes" or not. In theory, if attributes are supported, we should ALWAYS prefer them. Unfortunately, it's not that simple: even if you're using PHP 8, depending on your version of Doctrine & Symfony, your app might support the `Route` annotation but not the `Column` annotation. And so, I think it makes more sense to pass `$attributes` as an argument to each method where needed. So, similar to `addProperty()`, the method would have both a `$annotationLines` and `$attributes` arguments (we could remove/deprecate `$annotationLines` later). But unlike how you have it currently, we wouldn't try to be "smart" inside that method to choose one other the other: if `$annotationLines` is passed, we render those. Also, if `$attributes` is passed, we render those. The determination of which should be used would be done by whoever is calling that method.

Reply:
Since the `ClassSourceManipulator` isnt a Singleton or Service and is new initialized in every make command, the command itself could check for requirements.  E.g. the  `make:entity` command have the check if Symfony is >= 5.2 and doctrine is >= 2.8.    In that case you can keep your settings global as constructor parameter.

Well there is maybe another problem when 2 different packages need to have annotations/attributes on the same class and one of the packages does not support attributes. Let's take as an example Entities which are handled by Doctrine and ApiPlatform. When ApiPlatform does not support Attributes, you have to be able to pass `$attributes` and `$annotationLines` to make sure a that the caller of the method can decide which properties/methods should be annotated and which are attributed.

What do you think?

Reply:
That’s exactly the situation I was thinking of. So we should have no new constructor argument, but instead should allow both $annotationLines and $attributes to be passed to each method :)